### ISSUE TITLE
Clipboard content not preserved

### ISSUE THREAD

**Issue by L-U-T-i** [TigerVNC/tigervnc#1615]
**Describe the bug**
After I copy some text from a PDF file, paste it to Firefox on my PC first, then focus to a TigerVNC client window selecting some text in txt file there (to be replaced with the text from my PDF file on my PC) and choose 'paste', selected content from there (the text on TigerVNC server I've just selected, but not copied to a clipboard!) is pasted (instead of the text most recently copied to a clipboard, from PDF on my PC).

If I just copy the same text from my PC (TigerVNC client) and paste it over selected text on TigerVNC server (without pasting it on my PC / TigerVNC client first), everything works OK (I remember this has also been an issue in the past...).

**To Reproduce**
1. Select some text on a TigerVNC client machine
2. Paste this text on the same (TigerVNC client) machine
3. Select some text on TigerVNC server machine (to be replaced with the clipboard content from a client machine)
4. Right click 'Paste' over this text (or, choose 'Paste' action from a menu in text editor)

**Expected behavior**
Clipboard content (the text from a TigerVNC client machine) to be pasted over selected text on a TigerVNC server machine even if pasted before (on a TigerVNC client machine or some other TigerVNC server machine).

**Screenshots**
Are they needed?

**Client (please complete the following information):**
 - OS: Rocky Linux 9.1
 - VNC client: TigerVNC
 - VNC client version: 1.13.80 git snapshot 7a2af97 (08-03.2023), noticed the same behavior before with different released versions as well
 - Client downloaded from: custom build (based on CentOS / Fedora rpmbuild .spec)

**Server (please complete the following information):**
 - OS: Rocky Linux 9.1
 - VNC server: TigerVNC
 - VNC server version: 1.13.80 git snapshot 7a2af97 (08-03.2023)
 - Server downloaded from: custom build (based on CentOS / Fedora rpmbuild .spec)
 - Server was started using: systemd service

**Additional context**
I haven't noticed that before, with versions up to 1.12.0 (but can not confirm 100% that I've tried exactly that).

Built with the following configure options:
```
%configure \
        --disable-xorg --disable-xnest --disable-xvfb --disable-dmx \
        --disable-xwin --disable-xephyr --disable-kdrive --disable-xwayland \
        --with-pic --disable-static \
        --with-default-font-path="catalogue:%{_sysconfdir}/X11/fontpath.d,built-ins" \
        --with-fontrootdir=%{_datadir}/X11/fonts \
        --with-xkb-output=%{_localstatedir}/lib/xkb \
        --enable-glx --disable-dri --enable-dri2 --disable-dri3 \
        --disable-unit-tests \
        --disable-config-hal \
        --disable-config-udev \
        --without-dtrace \
        --disable-devel-docs \
        --disable-selective-werror
```

**Comment by CendioOssman** [comment]
Hmm... We fixed this scenario in 95ad4d7. Strange that you are still seeing issues.

I wonder if there is some clipboard manager in the client or server that is interacting poorly with TigerVNC.

Which desktop environments are you running on the client and server? Would it be possible to test something else to see if the problem goes away?

**Comment by L-U-T-i** [comment]
It is Rocky Linux 9.1 with Mate Desktop custom built (Fedora based .spec files, from the latest git commits). I am not noticing this issue pasting on client or server itself, only when VNC is involved.

But, as said, it seems to be different that in the past (what you mention). As much as I remember, it happened every time when the text has been selected then. Now, if I copy text, without pasting it somewhere, I can select some text on server and paste over it. The problem is only if I have already pasted the clipboard content somewhere else (on client desktop or in some other TigerVNC server) before.

**Comment by homerzhou** [comment]
same issue in Debian 11, tigervnc 1.12.0

**Comment by L-U-T-i** [comment]
I've noticed just today the following (if it helps anyhow to track down the clipboard weirdness):
1. I right-click the file in caja (file manager - mate desktop "nautilus" equivalent) , select "rename" and copy the file name into the clipboard (on a vnc client)
2. I try to paste this just copied name in mate-terminal on a vnc server (connection already established, just open the minimized window) - nothing happens, like the clipboard is empty, even if "paste" command is available in right-click menu (which is normally not when clipboard is empty!)
3. I can normally paste the filename on a vnc client (into a text editor or into a mate-terminal...) - before step 2 or after step 2, doesn't matter
4. If I paste the filename into a text editor on a vnc client first, and copy it (into the clipboard) after, I can normally paste it to the same mate-terminal on a vnc server (what was not possible before)

So, in short - copy-paste from a file browser window (on a vnc client) to a vnc server (anywhere - text editor, terminal...) doesn't work, while copy-paste from a text editor (on a vnc client) to a vnc server works at the same time normally.

**Comment by L-U-T-i** [comment]
At the same time, copy from a text editor on client and paste on server works, while cut the same text does not (paste is available in menu, but nothing is actually pasted).

**Comment by CendioOssman** [comment]
It sounds like this is a client issue, not a server issue, if it is relevant which application is used on the client side.

Do you have any other VNC server available that you could test and see if the issue is indeed in the client?

**Comment by samhed** [comment]
Closing due to no response.
